core/thread.py

```diff
@@ -474,67 +530,59 @@ def __getitem__(self, item):
             reason='Creating a thread channel'
         )
 
-        thread.channel = channel
+        thread = Thread(self, recipient, channel)
+        self.cache[recipient.id] = thread
 
-        log_url, log_data = await asyncio.gather(
-            self.bot.modmail_api.create_log_entry(recipient, channel, creator or recipient),
-            self.bot.modmail_api.get_user_logs(recipient.id)
-            # self.get_dominant_color(recipient.avatar_url)
-        )
+        log_url = await self.bot.api.create_log_entry(recipient, channel,
+                                                      creator or recipient)
+
+        log_data = await self.bot.api.get_user_logs(recipient.id)
+        # await self.get_dominant_color(recipient.avatar_url)
 
         log_count = sum(1 for log in log_data if not log['open'])
-        info_embed = self._format_info_embed(recipient, creator,
-                                             log_url, log_count,
+        info_embed = self._format_info_embed(recipient, log_url, log_count,
                                              discord.Color.green())
 
         topic = f'User ID: {recipient.id}'
-        mention = self.bot.config.get('mention', '@here') \
-            if not creator else None
+        if creator:
+            mention = None
+        else:
+            mention = self.bot.config.get('mention', '@here')
 
         _, msg = await asyncio.gather(
             channel.edit(topic=topic),
             channel.send(mention, embed=info_embed)
         )
 
         thread.ready = True
-
         await msg.pin()
-
         return thread
 
     async def find_or_create(self, recipient):
         return await self.find(recipient=recipient) or \
                await self.create(recipient)
 
     @staticmethod
-    def valid_image_url(url):
-        """Checks if a url leads to an image."""
-        types = ['.png', '.jpg', '.gif', '.webp']
-        parsed = urlparse(url)
-        if any(parsed.path.endswith(i) for i in types):
-            return url.replace(parsed.query, 'size=128')
-        return False
-
-    @asyncexecutor()
-    def _do_get_dc(self, image, quality):
-        with io.BytesIO(image) as f:
+    @async_executor()
+    def _do_get_dc(image, quality):
+        with BytesIO(image) as f:
             return ColorThief(f).get_color(quality=quality)
 
     async def get_dominant_color(self, url=None, quality=10):
```

Comment:
This is no longer necessary as its not being used anywhere

Reply:
Removed.